x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Concat.java

```diff
@@ -41,8 +41,8 @@ public class Concat extends EsqlScalarFunction {
     @FunctionInfo(returnType = "keyword", description = "Concatenates two or more strings.")
     public Concat(
         Source source,
-        @Param(name = "first", type = { "keyword", "text" }) Expression first,
-        @Param(name = "rest", type = { "keyword", "text" }) List<? extends Expression> rest
+        @Param(name = "string1", type = { "keyword", "text" }) Expression first,
+        @Param(name = "string2", type = { "keyword", "text" }) List<? extends Expression> rest
```

Comment:
```suggestion
        @Param(name = "firstString", type = { "keyword", "text" }) Expression first,
        @Param(name = "otherStrings", type = { "keyword", "text" }) List<? extends Expression> rest
```

Reply:
We cannot name this second parameter as `otherStrings` because both arguments are required. So it's "first required string" and the "second required string". The function is variadic and whatever arguments come after the first and second one are implicitly considered "the rest".

Reply:
With the proposed name this is the signature coming as output:
```
keyword concat(string1:keyword|text, string2...:keyword|text)
```

With my proposed changes it looks a bit more consistent to me:
```
keyword concat(firstString:keyword|text, otherStrings...:keyword|text)
```

Reply:
In terms of arguments, it looks to me that concat is similar to case, they both have two arguments, and the second one is a list and it is required. Shall we have a consistent naming convention for them? Use either plural or singular for the second argument's name? I don't have a strong opinion for either, just to make it a bit more consistent, I don't think either of them will be misleading.

Reply:
@fang-xing-esql use the singular version. The fact that `...` is present implies it is a list.